Src/PerformanceCollector/Filtering/Filter.cs

```diff
@@ -0,0 +1,596 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// Filter determines whether a telemetry document matches the criterion.
```

Comment:
Can you please include the expected configuration language and filtering login into the summary tag.